### PR TITLE
Remove forked chain when AppendBlocksAsync fail

### PR DESCRIPTION
**Pull request by earlbread**
This removes forked chain when `FillBlocksAsync` fail.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -151,7 +151,8 @@ To be released.
 [#522]: https://github.com/planetarium/libplanet/pull/522
 [#526]: https://github.com/planetarium/libplanet/pull/526
 [#527]: https://github.com/planetarium/libplanet/issues/527
-[#535]: https://github.com/planetarium/libplanet/issues/535
+[#537]: https://github.com/planetarium/libplanet/issues/537
```
Comment (moreal) [inline CHANGES.md]:
```suggestion
[#537]: https://github.com/planetarium/libplanet/pull/537
```

File: CHANGES.md [hunk 2]
```diff
@@ -151,7 +151,8 @@ To be released.
...
+[#540]: https://github.com/planetarium/libplanet/issues/540
```
Comment (moreal) [inline CHANGES.md]:
```suggestion
[#540]: https://github.com/planetarium/libplanet/pull/540
```

File: Libplanet/Net/Swarm.cs [hunk 3]
```diff
@@ -1597,29 +1597,43 @@ CancellationToken cancellationToken
...
+
+                    _logger.Debug("Sync is done.");
+                    if (!previousBlocks.Id.Equals(_blockChain.Id))
+                    {
+                        _logger.Debug("trying to swapping chain...");
+                        _blockChain.Swap(previousBlocks, render: true);
+                        _logger.Debug("Swapping complete");
+                    }
+                }
+                catch (Exception)
```
Comment (longfin) [inline Libplanet/Net/Swarm.cs]: How about converting this block to `finally`?

Comment (earlbread) [reply Libplanet/Net/Swarm.cs]: Changed!

File: Libplanet/Net/Swarm.cs [hunk 4]
```diff
@@ -1597,29 +1597,41 @@ CancellationToken cancellationToken
...
+                        _logger.Debug("Swapping complete");
+                    }
+                }
+                finally
                 {
-                    _logger.Debug("trying to swapping chain...");
-                    _blockChain.Swap(previousBlocks, render: true);
-                    _logger.Debug("Swapping complete");
+                    Guid? canonicalChainId = _blockChain.Store.GetCanonicalChainId();
+                    if (previousBlocks != null && canonicalChainId != previousBlocks.Id)
```
Comment (longfin) [inline Libplanet/Net/Swarm.cs]:
```suggestion
                    if (canonicalChainId != previousBlocks?.Id)
```

**Pull request merged**